Bottom: b141cd6513808a3d38f0fbbd29863c78b625ae79
Top:    a201b0e328ce25b5ffa45b002cb438d281462a5c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-11-22 18:28:36 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 3f68165..5318a52 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2610,7 +2610,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
